### PR TITLE
Move travis phpunit to github actions

### PR DESCRIPTION
**Pull request by skjnldsv**
Cleaner pr :)

### CODE REVIEW

File: .github/workflows/phpunit.yml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        server-versions: ['master']
+  
+    name: php${{ matrix.php-versions }}/${{ matrix.databases }} on ${{ matrix.server-versions }}
+
+    steps:
+      - name: Checkout server
+        uses: actions/checkout@v2
+        with:
+          repository: nextcloud/server
+          ref: test/debug-install
```
Comment (skjnldsv) [inline .github/workflows/phpunit.yml]:
```suggestion
          ref: ${{ matrix.server-versions }}
```

Comment (skjnldsv) [reply .github/workflows/phpunit.yml]: We could even use [`github.ref`](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#github-context)

Comment (juliushaertl) [reply .github/workflows/phpunit.yml]: I'd say we keep it like this for now, as a lot of apps do not have the same stable branches like the server does. We can add a comment about that, so we can change it to base_ref for shipped apps.

**Comment by skjnldsv** [comment]
I'm not fond of the naming though, because it create lots of different settings in the branches protections and it means any update, like php 7.5, the list will become super annoying
image

**Comment by ChristophWurst** [comment]
IMO the matrix is too big. You don't have to run every possible combination. For a real-world app a large matrix will result in slower CI runs, while lots of the executed tests are redundant.

**Comment by skjnldsv** [comment]
> IMO the matrix is too big. You don't have to run every possible combination. For a real-world app a large matrix will result in slower CI runs, while lots of the executed tests are redundant.

Sure :)
What is really necessary or needed?

**Comment by ChristophWurst** [comment]
> What is really necessary or needed?

I usually have all tests run for each php version, them one for each database and one for each major release of Nextcloud. But not mixed :)

**Comment by skjnldsv** [comment]
> I usually have all tests run for each php version, them one for each database and one for each major release of Nextcloud. But not mixed :)

Which database you choose then?

**Comment by ChristophWurst** [comment]
MySQL by default. And two more runs for sqlite and postgres

**Comment by skjnldsv** [comment]
> MySQL by default. And two more runs for sqlite and postgres

What about the naming of the runs in the branch protection?
Shall we go for a more standard one?

**Comment by ChristophWurst** [comment]
> What about the naming of the runs in the branch protection?

Be creative

> Shall we go for a more standard one?

Just find a good one. People are free to name their CI runs however they like. This repo's purpose is just to get them started.

**Comment by splitt3r** [comment]
Some very small nitpicks: unify formatting with an empty line before each `name:` (makes it more readable in my oprionion espacily for beginners). For the naming scheme i like kebab case notation like php7.4-mysql-17. In general a great addition!

**Comment by skjnldsv** [comment]
@splitt3r can you provide the changes? :)

**Comment by splitt3r** [comment]
yes i will 👍

**Comment by splitt3r** [comment]
PHPUnit 9 incoming 🚀

**Comment by skjnldsv** [comment]
> PHPUnit 9 incoming

Just saw that, let's use our composer phpunit

**Pull request merged**